src/MLJLIBSVMInterface.jl

```diff
@@ -44,83 +47,97 @@ function LinearSVC(
     return model
 end
 
-mutable struct SVC <: MMI.Deterministic
-    kernel
-    gamma::Float64
-    cost::Float64
-    cachesize::Float64
-    degree::Int32
-    coef0::Float64
-    tolerance::Float64
-    shrinking::Bool
-    probability::Bool
-end
-
-function SVC(
-    ;kernel = LIBSVM.Kernel.RadialBasis
-    ,gamma::Float64 = 0.0
-    ,cost::Float64 = 1.0
-    ,cachesize::Float64=200.0
-    ,degree::Int32 = Int32(3)
-    ,coef0::Float64 = 0.0
-    ,tolerance::Float64 = .001
-    ,shrinking::Bool = true
-    ,probability::Bool = false)
-
-    model = SVC(
-        kernel
-        ,gamma
-        ,cost
-        ,cachesize
-        ,degree
-        ,coef0
-        ,tolerance
-        ,shrinking
-        ,probability
-    )
-
-    message = MMI.clean!(model)   #> future proof by including these
-    isempty(message) || @warn message #> two lines even if no clean! defined below
-
-    return model
+for Model in [:SVC, :ProbabilisticSVC]
+
+    SuperType = Model === :SVC ? :Deterministic : :Probabilistic
+
+    quote
+        mutable struct $Model <: MMI.$SuperType
+            kernel
+            gamma::Float64
+            cost::Float64
+            cachesize::Float64
+            degree::Int32
+            coef0::Float64
+            tolerance::Float64
+            shrinking::Bool
+            probability::Bool
+        end
+
+        function $Model(
+            ;kernel = LIBSVM.Kernel.RadialBasis
+            ,gamma::Float64 = 0.0
+            ,cost::Float64 = 1.0
+            ,cachesize::Float64=200.0
+            ,degree::Int32 = Int32(3)
+            ,coef0::Float64 = 0.0
+            ,tolerance::Float64 = .001
+            ,shrinking::Bool = true
+            ,probability::Bool = false)
```

Comment:
We shouldn't add the `probability` kwarg here. Since we don't want the user to change it.
The probability argument is created in the `svmtrain` method for `Deterministic` and `Probabilitic` SVM models

Reply:
Good catch.